crates/uv/tests/common/mod.rs

```diff
@@ -168,10 +172,35 @@ impl TestContext {
     ///
     /// See [`TestContext::new`] if only a single version is desired.
     pub fn new_with_versions(python_versions: &[&str]) -> Self {
-        let temp_dir = assert_fs::TempDir::new().expect("Failed to create test working directory");
-        let cache_dir = assert_fs::TempDir::new().expect("Failed to create test cache directory");
-        let python_dir = assert_fs::TempDir::new().expect("Failed to create test Python directory");
-        let home_dir = assert_fs::TempDir::new().expect("Failed to create test home directory");
+        // Discover the path to the XDG state directory. We use this, rather than the OS-specific
+        // temporary directory, because on macOS (and Windows on GitHub Actions), they involve
+        // symlinks. (On macOS, the temporary directory is, like `/var/...`, which resolves to
+        // `/private/var/...`.)
+        //
+        // It turns out that, at least on macOS, if we pass a symlink as `current_dir`, it gets
+        // _immediately_ resolved (such that if you call `current_dir` in the running `Command`, it
+        // returns resolved symlink). This is problematic, as we _don't_ want to resolve symlinks
+        // for user-provided paths.
+        let bucket = etcetera::base_strategy::choose_base_strategy()
+            .expect("Failed to find base strategy")
+            .data_dir()
+            .join("uv")
+            .join("tests");
```

Comment:
Sorry, one more ask here — can we respect `UV_INTERNAL__TEST_DIR` or something so downstream maintainers can change this if they need to?